components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -864,6 +939,17 @@ private static boolean processTokenRevocation(Set<String> clientIds, Authenticat
         return isErrorOnRevokingTokens;
     }
 
+    private static String resolveRootTenantDomain(String tenantDomain) {
+
+        try {
+            String rootOrgId = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                    .getPrimaryOrganizationId(tenantDomain);
+            return OAuthComponentServiceHolder.getInstance().getOrganizationManager().resolveTenantDomain(rootOrgId);
+        } catch (OrganizationManagementException e) {
+            throw new RuntimeException(e);
```

Comment:
make this a checked exception

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2382/commits/3a73357862da03385d0d13f543d385b492fb314d